### PR TITLE
paperwork: Add version 2.0

### PR DESCRIPTION
**Pull request by akuntsch**
`checkver` is a bit ugly because the jsonpath implementation does not allow selecting multiple fields which would make the regex simpler. Currently it heavily relies on the format of the Gitlab API response.

### CODE REVIEW

File: bucket/paperwork.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "version": "2.0",
+    "description": "Personal document manager",
+    "homepage": "https://openpaper.work",
+    "license": "GPL-3.0-only",
```
Comment (Ash258) [inline bucket/paperwork.json:5]: A

```suggestion
    "license": "GPL-3.0-or-later",
```

File: bucket/paperwork.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "url": "https://download.openpaper.work/windows/x86/paperwork-master-24ba2639.zip",
+    "hash": "85414f4789e000b60ede495df8e314c7328d47ae16a163c1110e0d1a0b9cbdcd",
+    "shortcuts": [
+        [
+            "paperwork.exe",
+            "Paperwork",
+            "",
+            "data/paperwork_64.ico"
```
Comment (Ash258) [inline bucket/paperwork.json:13]:
```suggestion
            "data\\paperwork_64.ico"
```

File: bucket/paperwork.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        ]
+    ],
+    "checkver": {
+        "url": "https://gitlab.gnome.org/api/v4/projects/World%2FOpenPaperwork%2Fpaperwork/repository/tags?order_by=name",
```
Comment (Ash258) [inline bucket/paperwork.json:17]: A

Go through the source code/reverse engineer and provide native version check.

**Review by Ash258 — CHANGES_REQUESTED** [review]
I am against adding application without releases to the official buckets. The checkver will not be executed until there is global release of the application. Leading to the potential of purging old installers/build artifacts and download failures.

File: bucket/paperwork.json (hunk 3 above)
Comment (akuntsch) [reply bucket/paperwork.json:17]: https://openpaper.work/beacon/latest is used for the update check. This gives me the version but for the download URL I need the commit hash.

**Comment by akuntsch** [comment]
> I am against adding application without releases to the official buckets. The checkver will not be executed until there is global release of the application. Leading to the potential of purging old installers/build artifacts and download failures.

What do you mean by releases? Releases which would show up at https://gitlab.gnome.org/World/OpenPaperwork/paperwork/-/releases or do release git tags count?

For the version I just went with what Arch Linux uses (they build the package from source obviously): https://github.com/archlinux/svntogit-community/blob/packages/paperwork/trunk/PKGBUILD#L14

Even with a release in their GitLab repo the problem of obtaining the download URL would probably persist because past releases did not contain compiled assets and seemingly the only file under https://download.openpaper.work/windows without commit hash in its name seems to be https://download.openpaper.work/windows/amd64/paperwork-master-1.2.4.zip.

Not sure if the same kind of file will be available for 2.0 in the future.

**Review by issaclin32 — APPROVED** [review]

**Pull request merged**